proposals/2024-04-10-native-support-for-info-metrics-metadata.md

```diff
@@ -0,0 +1,112 @@
+# Add 1st class feature to PromQL for handling info type metrics
+
+* **Owners:**
+  * Arve Knudsen [@aknuds1](https://github.com/aknuds1) [arve.knudsen@grafana.com](mailto:arve.knudsen@grafana.com)
+
+* **Implementation Status:** Partially implemented
+
+* **Related Issues and PRs:**
+  * [WIP: Info PromQL function prototype](https://github.com/grafana/mimir-prometheus/pull/598)
+
+* **Other docs or links:**
+  * [Proper support for OTEL resource attributes](https://docs.google.com/document/d/1FgHxOzCQ1Rom-PjHXsgujK8x5Xx3GTiwyG__U3Gd9Tw/edit#heading=h.unv3m5m27vuc)
+  * [Special treatment of info metrics in Prometheus](https://docs.google.com/document/d/1ebhGNLs3uhdeprJCullM-ywA9iMRDg_mmnuFAQCloqY/edit#heading=h.2rmzk7oo6tu8)
+  * [Scenarios scratch pad](https://docs.google.com/document/d/1nV6N3pDfvZhmG2658huNbFSkz2rsM6SpkHabp9VVpw0/edit#heading=h.luf3yapzr29e)
+
+> This proposal collects the requirements and implementation proposals for adding a 1st class feature to PromQL for handling info type metrics.
+
+## Why
+
+Currently, enriching Prometheus query results with corresponding labels from info metrics is challenging.
```

Comment:
Can we introduce info metric first? Or is it just a gauge now?

Reply:
Exactly yes!

Reply:
It would certainly be possible to introduce the necessary info metric modeling before the `info` function, but I think first of all we have to agree on the model :) The `info` function's semantics helps us understand what kind of functionality the model must provide.

Reply:
Makes sense @beorn7, I'll try to add to the introduction as you suggest.

Reply:
Done @beorn7 - PTAL.